homeassistant/components/unifi/device_tracker.py

```diff
@@ -62,111 +65,146 @@
+            controller, async_add_entities, request_update, tracked
+        )]
+        await asyncio.wait(tasks)
+
+    await update_controller()
+
+
+async def async_update_items(
+        controller, async_add_entities, request_controller_update, tracked):
+    """Update POE port state from the controller."""
     try:
-        controller = await get_controller(
-            hass, host, username, password, port, site_id, verify_ssl)
-        await controller.initialize()
-
-    except (AuthenticationRequired, CannotConnect) as ex:
-        _LOGGER.error("Failed to connect to Unifi: %s", ex)
-        hass.components.persistent_notification.create(
-            'Failed to connect to Unifi. '
-            'Error: {}<br />'
-            'You will need to restart hass after fixing.'
-            ''.format(ex),
-            title=NOTIFICATION_TITLE,
-            notification_id=NOTIFICATION_ID)
-        return False
-
-    return UnifiScanner(
-        controller, detection_time, ssid_filter, monitored_conditions)
-
-
-class UnifiScanner(DeviceScanner):
-    """Provide device_tracker support from Unifi WAP client data."""
-
-    def __init__(self, controller, detection_time: timedelta,
-                 ssid_filter, monitored_conditions) -> None:
-        """Initialize the scanner."""
+        with async_timeout.timeout(4):
+            await controller.api.clients.update()
+            await controller.api.devices.update()
+
+    except aiounifi.LoginRequired:
+        try:
+            with async_timeout.timeout(5):
+                await controller.api.login()
+
+        except (asyncio.TimeoutError, aiounifi.AiounifiException):
+            if controller.available:
+                controller.available = False
+                # update_switch_state()
+            return
+
+    except (asyncio.TimeoutError, aiounifi.AiounifiException):
+        if controller.available:
+            LOGGER.error('Unable to reach controller %s', controller.host)
+            controller.available = False
+            # update_switch_state()
+        return
+
+    if not controller.available:
+        LOGGER.info('Reconnected to controller %s', controller.host)
+        controller.available = True
+
+    new_tracked = []
+    for client_id in controller.api.clients:
+
+        if client_id in tracked:
+            LOGGER.debug("Updating UniFi tracked device %s (%s)",
+                         tracked[client_id].entity_id,
+                         tracked[client_id].client.mac)
+            tracked[client_id].async_schedule_update_ha_state()
+            continue
+
+        client = controller.api.clients[client_id]
+
+        tracked[client_id] = UniFiTracker(
+            client, controller, request_controller_update)
+        new_tracked.append(tracked[client_id])
+        LOGGER.debug("New UniFi switch %s (%s)", client.hostname, client.mac)
+
+    if new_tracked:
+        async_add_entities(new_tracked)
+
+
+class UniFiTracker(DeviceTrackerEntity):
```

Comment:
You will need to create a new DeviceTrackerEntity base class for entities that are being seen on a router. We should not merge that logic with the GPS based entity, it's a different data set. This new entity could also include automatic updates when last_seen passes a certain limit.